recipes-browser/chromium/chromium/chromium-wayland/0001-Fix-wayland-window-show-hide.patch

```diff
@@ -0,0 +1,87 @@
+From 3cd3f4d5beeea7abadc00ebfb39247fe0a0eef50 Mon Sep 17 00:00:00 2001
+From: Maksim Sisov <msisov@igalia.com>
+Date: Thu, 8 Jun 2017 13:37:08 +0300
+Subject: [PATCH] Fix wayland window show/hide
+
+---
+ ui/ozone/platform/wayland/wayland_window.cc | 27 ++++++++++++++++++++-------
+ ui/ozone/platform/wayland/wayland_window.h  |  2 ++
+ 2 files changed, 22 insertions(+), 7 deletions(-)
+
+diff --git a/ui/ozone/platform/wayland/wayland_window.cc b/ui/ozone/platform/wayland/wayland_window.cc
+index 6775c75..fb8c473 100644
+--- a/ui/ozone/platform/wayland/wayland_window.cc
++++ b/ui/ozone/platform/wayland/wayland_window.cc
+@@ -52,13 +52,9 @@ WaylandWindow* WaylandWindow::FromSurface(wl_surface* surface) {
+ }
+ 
+ bool WaylandWindow::Initialize() {
+-  surface_.reset(wl_compositor_create_surface(connection_->compositor()));
+-  if (!surface_) {
+-    LOG(ERROR) << "Failed to create wl_surface";
++  CreateSurfaceAndSetUserData();
++  if (!surface_)
+     return false;
+-  }
+-  wl_surface_set_user_data(surface_.get(), this);
+-
+   // There is now default initialization for this type. Initialize it
+   // to ::WINDOW here. It will be changed by delelgate if it know the
+   // type of the window.
+@@ -111,6 +107,10 @@ void WaylandWindow::ApplyPendingBounds() {
+ 
+ void WaylandWindow::CreatePopupWindow() {
+   DCHECK(parent_window_);
++  CreateSurfaceAndSetUserData();
++  if (!surface_)
++    return;
++
+   gfx::Rect bounds =
+       TranslateBoundsToScreenCoordinates(bounds_, parent_window_->GetBounds());
+   xdg_popup_.reset(xdg_shell_get_xdg_popup(
+@@ -119,6 +119,16 @@ void WaylandWindow::CreatePopupWindow() {
+   xdg_popup_add_listener(xdg_popup_.get(), &xdg_popup_listener, this);
+ }
+ 
++void WaylandWindow::CreateSurfaceAndSetUserData() {
```

Comment:
I would change this method to return a bool.

on the call sites:

if (!CreateSurfaceAndSetUserData())
  return;